### DISCUSSION TITLE
Help regarding a few errors when saving web pages

### DISCUSSION THREAD

**Discussion by blogdemoi** [ArchiveBox/ArchiveBox#1148] [Q&A]
Hi,

I am quite happy with my Archivebox, on a Raspberry Pi 3B, DietPi OS, saving to an attached usb key.

Sometimes I have errors. Today I tried to save a web page, from the Gui first, but pdf, single file and medias where not saved.
So I tried to save from the cli and I also have the same errors:

> dietpi@DietPi:~/archivebox$ docker compose run archivebox add 'https://stephenjones.blog/2023/05/07/xu-tong/'
> [i] [2023-05-07 13:43:09] ArchiveBox v0.6.2: archivebox add https://stephenjones.blog/2023/05/07/xu-tong/
>     > /data
> 
> [+] [2023-05-07 13:43:20] Adding 1 links to index (crawl depth=0)...
>     > Saved verbatim input to sources/1683467000-import.txt
>     > Parsed 1 URLs from input (Generic TXT)                                    
>     > Found 1 new URLs not already in index
> 
> [*] [2023-05-07 13:43:20] Writing 1 links to main index...
>     √ ./index.sqlite3                                                           
> 
> [▶] [2023-05-07 13:43:20] Starting archiving of 1 snapshots in index...
> 
> [+] [2023-05-07 13:43:22] "stephenjones.blog/2023/05/07/xu-tong"
>     https://stephenjones.blog/2023/05/07/xu-tong/
>     > ./archive/1683467000.141085
>       > title
>       > favicon                                                                 
>       > headers                                                                 
>       > singlefile                                                              
>         Extractor failed:                                                       
>              SingleFile was not able to archive the page
>             Got single-file response code: 0.
>             TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
>         Run to see full output:
>             cd /data/archive/1683467000.141085;
>             /node/node_modules/single-file/cli/single-file --browser-executable-path=chromium "--browser-args=[\"--headless\", \"--no-sandbox\", \"--disable-gpu\", \"--disable-dev-shm-usage\", \"--disable-software-rasterizer\", \"--run-all-compositor-stages-before-draw\", \"--hide-scrollbars\", \"--user-agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.61 Safari/537.36 ArchiveBox/{VERSION} (+https://github.com/ArchiveBox/ArchiveBox/)\", \"--window-size=1440,2000\"]" https://stephenjones.blog/2023/05/07/xu-tong/ singlefile.html
> 
>       > pdf
>         Extractor timed out after 60s.                                          
>         Run to see full output:
>             cd /data/archive/1683467000.141085;
>             chromium --headless --no-sandbox --disable-gpu --disable-dev-shm-usage --disable-software-rasterizer --run-all-compositor-stages-before-draw --hide-scrollbars "--user-agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.61 Safari/537.36 ArchiveBox/{VERSION} (+https://github.com/ArchiveBox/ArchiveBox/)" --window-size=1440,2000 --timeout=60000 --print-to-pdf https://stephenjones.blog/2023/05/07/xu-tong/
> 
>       > screenshot
>       > dom                                                                     
>       > wget                                                                    
>         Extractor timed out after 60s.                                          
>         Run to see full output:
>             cd /data/archive/1683467000.141085;
>             wget --no-verbose --adjust-extension --convert-links --force-directories --backup-converted --span-hosts --no-parent -e robots=off --timeout=60 --restrict-file-names=windows --warc-file=/data/archive/1683467000.141085/warc/1683467143 --page-requisites "--user-agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.61 Safari/537.36 ArchiveBox/0.6.2 (+https://github.com/ArchiveBox/ArchiveBox/) wget/GNU Wget 1.20.1" --compression=auto https://stephenjones.blog/2023/05/07/xu-tong/
> 
>       > readability
>       > mercury                                                                 
>       > media                                                                   
>         Extractor failed:                                                       
>              Failed to save media
>             Got youtube-dl response code: 1.
>             WARNING: Falling back on generic information extractor.
>             ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
>             ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
>             ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
>         Run to see full output:
>             cd /data/archive/1683467000.141085;
>             youtube-dl --write-description --write-info-json --write-annotations --write-thumbnail --no-call-home --write-sub --all-subs --write-auto-sub --convert-subs=srt --yes-playlist --continue --ignore-errors --geo-bypass --add-metadata --max-filesize=750m https://stephenjones.blog/2023/05/07/xu-tong/
> 
>       > archive_org
>         Extractor timed out after 60s.                                          
>         Run to see full output:
>             cd /data/archive/1683467000.141085;
>             curl --silent --location --compressed --head --max-time 60 --user-agent "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.61 Safari/537.36 ArchiveBox/0.6.2 (+https://github.com/ArchiveBox/ArchiveBox/) curl/curl 7.64.0 (aarch64-unknown-linux-gnu)" https://web.archive.org/save/https://stephenjones.blog/2023/05/07/xu-tong/
> 
>         110 files (8.3 MB) in 0:05:48s 
> 
> [√] [2023-05-07 13:49:10] Update of 1 pages complete (5.84 min)
>     - 0 links skipped
>     - 1 links updated
>     - 1 links had errors
> 
>     Hint: To manage your archive in a Web UI, run:
>         archivebox server 0.0.0.0:8000

Could you advise?

Thank you.

**Comment by pirate** [comment]
It's normal to see occasional errors for individual URLs and extractor methods, that's why we try many methods for each URL. You can always try updating the extractor dependencies listed in `archivebox version` if you're on a bare-metal install, or trying the latest `archivebox/archivebox:dev` docker image in case it's an extractor thats out of date (like SingleFile or yt-dlp).

If you see recurring errors for more than 20~30% of specific URL extractor types then it's worth investigating further.
